### PR TITLE
Fix using custom formatters

### PR DESCRIPTION
**Pull request by xDimon**
Some custom formatters do not work with logger, because logger used fmt::format_to_t specified by pointer which does not use custom formatter. Replacing pointer by iterator resolves that.

### CODE REVIEW

File: include/soralog/event.hpp [hunk 1]
```diff
@@ -57,12 +57,33 @@ namespace soralog {
...
+        value_type *pos;
+
+        value_type &operator*() const {
+          return *pos;
+        }
+        constexpr auto &operator++() {
+          pos++;
+          return *this;
+        }
+        constexpr auto &operator++(int n) {
```
Comment (turuslan) [inline include/soralog/event.hpp]: Is this `it++`?
```suggestion
        constexpr auto &operator++(int) {
```
Or is it `it += n`
```suggestion
        constexpr auto &operator+=(int n) {
```

Comment (xDimon) [reply include/soralog/event.hpp]: First one. Fix pushed

Comment (igor-egorov) [reply include/soralog/event.hpp]: seems like it should be +=

**Review by igor-egorov — APPROVED** [review]
Please fix Ruslan's note

**Review by turuslan — APPROVED** [review]

**Pull request merged**